### ISSUE TITLE
"Unable to parse fact" exception for git dependencies

### ISSUE THREAD

**Issue by desophos** [python-poetry/poetry#5334]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 10 21H2
- **Poetry version**: 1.2.0b1
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/desophos/9490a9fd5dc4b1daec311baafb3ebcc5

## Issue

With Python 3.10.2 and Poetry 1.2.0b1, `poetry update` and `poetry lock` raise a `ValueError` when attempting to parse a git dependency: 
```
Unable to parse fact: fact: poetry-example depends on chromecontroller ()
``` 
with `chromecontroller` substituted for any git dependency. I tested it with https://github.com/desophos/telenium as well. 

The output of `poetry update -vvv` is included in the Gist.

**Comment by paveldroo** [comment]
Same problem here with the same version of poetry.
`Unable to parse fact: fact: some_poetry_app depends on keras_contrib ()`

**Comment by abn** [comment]
The root cause for this issue are the following lines.

https://github.com/python-poetry/poetry-core/blob/f405311de58922dcd0b42f792ced5801ee344960/src/poetry/core/packages/vcs_dependency.py#L100-L101

This is used at https://github.com/python-poetry/poetry/blob/3780a727f3bfebed99c68c262c53348fa12e7eb7/src/poetry/mixology/incompatibility.py#L443-L450

This leads to the string generated when logging the incompatibility. You can workaround this issue by simply adding a rev/branch/tag in your `pyproject.toml`

```sh
poetry add git+https://github.com/desophos/ChromeController.git#master
```

We should fix the default pretty constraint in core to return `rev HEAD`  and also ensure the regex in provider is fixed to handle empty values when logging.

https://github.com/python-poetry/poetry/blob/3780a727f3bfebed99c68c262c53348fa12e7eb7/src/poetry/puzzle/provider.py#L723

**Comment by abn** [comment]
@desophos also, jfyi the version used in `build-system` for `poetry-core` in your `pyproject.toml` is incorrect. Latest available version is [`1.1.0a7`](https://github.com/python-poetry/poetry-core/releases).
